config/configgrpc/configgrpc.go

```diff
@@ -181,7 +182,17 @@ func (gcs *GRPCClientSettings) ToClientConn(ctx context.Context, host component.
 		return nil, err
 	}
 	opts = append(opts, extraOpts...)
-	return grpc.DialContext(ctx, gcs.SanitizedEndpoint(), opts...)
+
+	// gRPC round_robin is only enabled when the DNS resolver is used.
+	// There are 2 options to use it.
+	// This function takes #2.
+	// 1. configure the global resolver by `resolver.SetDefaultScheme("dns")`
+	// 2. add "dns:[//authority/]" prefix in grpc.DialContext
+	var endpoint string = gcs.SanitizedEndpoint()
+	if gcs.BalancerName == roundrobin.Name {
+		endpoint = "dns:" + endpoint
```

Comment:
Why should this be hardcoded instead of user-provided in the config? What if another user wants to use a different resolver?

Reply:
@yurishkuro, thank you for your comment! I confirmed the following config let Traefik use the DNS resolver and round_robin option worked. 

```yaml
exporters:
  otlp:
    endpoint: "dns://foo.com:4317"
  balancer_name: round_robin
```

Because it looks changing from `loadBalancingPolicy` to `loadBalancingConfig` is not required yet, I'll close the PR.

Reply:
correction.
```yaml
exporters:
  otlp:
    endpoint: "dns:///foo.com:4317"
  balancer_name: round_robin
```

or
```yaml
exporters:
  otlp:
    endpoint: "dns:foo.com:4317"
  balancer_name: round_robin
```